### PR TITLE
Add pattern for writing and creating man pages for the splinter cli

### PR DESCRIPTION
**Pull request by agunde406**
This PR lays out how we can write documentation for cli commands. Each man page will have a markdown file in the provided TEMPLATE format. The markdown file will be converted to a man page with pandoc when the cli is built (script in build.rs file) and put in package/man. When the cli deb is built these files will be included in /user/share/man/man1. 

The build.rs file requires the pandoc is installed. If you do not want to generate the man pages set 
`export SPLINTER_NO_MAN_PAGES=true` If SPLINTER_NO_MAN_PAGES is set to false or does not exist the man pages will try to be generated.

The docker files for spllinterd and splinter cli have been updated to be able generate the man pages (installed pandoc) and installs the man pages in the containers (installed man and run mandb).

To test build splinterd or the splinter cli docker images and run. Docker exec into the container and run `man splinter-cert-generate` to see the man page for `splinter cert generate`

### CODE REVIEW

File: cli/man/splinter-cert-generate.1.md [hunk 1]
```diff
@@ -0,0 +1,95 @@
+% SPLINTER-CERT-GENERATE(1) Cargill, Incorporated | Splinter Commands
+
+NAME
+====
+
+**splinter cert generate** — Generates test certificates and keys for running
```
Comment (chenette) [inline cli/man/splinter-cert-generate.1.md]: Please change to splinter-cert-generate.  (This is a style change from my earlier recommendation, as noted in my comment on `cli/man/TEMPLATE.1.md.example`.) Thanks, and sorry for the extra work.

File: cli/man/TEMPLATE.1.md.example [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+<!--
+This file should be named the full command (with dashes between subcommands)
+with an extension of `1.md`. For example `splinter-cert-generate.1.md` is the
+markdown file that will generated into the man page for
+`splinter generate cert`.
+-->
+NAME
+====
+
+**command-name** — about from command help
```
Comment (chenette) [inline cli/man/TEMPLATE.1.md.example]: I suggest changing "command-name" to "command-subcommand" to clarify the use of dashes.  

Note: Using dashes here is a change from my earlier recommendation. I originally requested spaces between subcommands, but I have changed my mind after researching how similar commands present this line. I'd like to follow the git model, where the command-subcommand string on this line shows what someone would type to view the man page.

Comment (chenette) [reply cli/man/TEMPLATE.1.md.example]: Also (minor nitpick): I'd suggest writing "About blurb from command help" to reinforce that the blurb should start with a capital letter.

File: cli/man/TEMPLATE.1.md.example [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+the Splinter environment, provides big-picture information on who uses it and
+why (or when), and lists prerequisites or other considerations.
+
+FLAGS
+=====
+<!--
+List of flags and options (in alphabetical order) that describes what it does,
+what arguments it takes, and the default value/behavior. If appropriate, explain
+how/when it's used, with tips, gotchas, and very short examples.
+-->
```
Comment (chenette) [inline cli/man/TEMPLATE.1.md.example]: To clarify the expected info in this section, this comment should be for the flags only (they don't have arguments, for example). My suggestion:

```
List of flags (in alphabetical order) that describes what each flag does 
and the default value/behavior. If appropriate, explain
how/when it's used, with tips, gotchas, and very short examples.
```

File: cli/man/TEMPLATE.1.md.example [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+
+-V, --version
+: Prints version information
+
+-v
+: Increases verbosity (the opposite of -q). Specify multiple times for more
+  output.
+
+OPTIONS
+=======
```
Comment (chenette) [inline cli/man/TEMPLATE.1.md.example]: Add an option-specific comment like this:

```
List of options (in alphabetical order) that describes what the option does,
what arguments it takes, and the default value/behavior. If appropriate, explain
how/when it's used, with tips, gotchas, and very short examples.
```

File: cli/man/TEMPLATE.1.md.example [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+-e, --example <example>
+: Description about example options
+
+
+EXAMPLES
+========
+Show simple uses first, followed by more complex examples.
```
Comment (chenette) [inline cli/man/TEMPLATE.1.md.example]: Should be a comment.  Also, perhaps we should add something like "For a very simple subcommand where an example isn't appropriate, delete the EXAMPLES section."

File: cli/man/TEMPLATE.1.md.example [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+
+ENVIRONMENT
+===========
+The following environment variables affect the execution of the command.
+
+<!--
+Add comments to see
+-->
```
Comment (chenette) [inline cli/man/TEMPLATE.1.md.example]: Change to something like: "List each environment variable with a description; mention any related flags or options."

File: cli/man/TEMPLATE.1.md.example [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+
+**EXAMPLE_ENV_VAR**
+
+: Description of environment variable.  (See `--example`)
+
+SEE ALSO
+========
+Related man pages or documentation (if possible)
```
Comment (chenette) [inline cli/man/TEMPLATE.1.md.example]: Should be a comment.

File: cli/man/TEMPLATE.1.md.example [hunk 8]
```diff
@@ -0,0 +1,69 @@
+% COMMAND(1) Cargill, Incorporated | Splinter Commands
+
+<!--
+This file should be named the full command (with dashes between subcommands)
```
Comment (chenette) [inline cli/man/TEMPLATE.1.md.example]: To clarify, I suggest changing to:

"Template for a man page source file in Markdown format. The  file should have the name of the command plus any subcommands (with dashes ...)

File: cli/man/TEMPLATE.1.md.example [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+with an extension of `1.md`. For example `splinter-cert-generate.1.md` is the
+markdown file that will generated into the man page for
```
Comment (chenette) [inline cli/man/TEMPLATE.1.md.example]: Please change to "that generates the man page for..."

Also (nitpick): Markdown is written with a capital M.

File: cli/man/TEMPLATE.1.md.example [hunk 10]
```diff
@@ -0,0 +1,69 @@
...
+
+**EXAMPLE_ENV_VAR**
+
+: Description of environment variable.  (See `--example`)
+
+SEE ALSO
+========
+Related man pages or documentation (if possible)
+
+For more information see the Splinter documentation at
```
Comment (chenette) [inline cli/man/TEMPLATE.1.md.example]: Please add a comma after "information"

File: cli/man/splinter-cert-generate.1.md [hunk 11]
```diff
@@ -0,0 +1,95 @@
...
+generate:
+
+**SPLINTER_CERT_DIR**
+
+: The certificates and keys will be generated at the location specified by the
+  environment variable.  (See `--cert-dir`)
+
+SEE ALSO
+========
+For more information see the Splinter documentation at
```
Comment (chenette) [inline cli/man/splinter-cert-generate.1.md]: Add a comma after "information"

File: cli/man/splinter-cert-generate.1.md (hunk 1 above)
Comment (agunde406) [reply cli/man/splinter-cert-generate.1.md]: updated

File: cli/man/TEMPLATE.1.md.example (hunk 2 above)
Comment (agunde406) [reply cli/man/TEMPLATE.1.md.example]: updated

File: cli/man/TEMPLATE.1.md.example (hunk 3 above)
Comment (agunde406) [reply cli/man/TEMPLATE.1.md.example]: updated

File: cli/man/TEMPLATE.1.md.example (hunk 4 above)
Comment (agunde406) [reply cli/man/TEMPLATE.1.md.example]: updated

File: cli/man/TEMPLATE.1.md.example (hunk 5 above)
Comment (agunde406) [reply cli/man/TEMPLATE.1.md.example]: updated

File: cli/man/TEMPLATE.1.md.example (hunk 6 above)
Comment (agunde406) [reply cli/man/TEMPLATE.1.md.example]: updated

File: cli/man/TEMPLATE.1.md.example (hunk 7 above)
Comment (agunde406) [reply cli/man/TEMPLATE.1.md.example]: updated

File: cli/man/TEMPLATE.1.md.example (hunk 8 above)
Comment (agunde406) [reply cli/man/TEMPLATE.1.md.example]: updated

File: cli/man/TEMPLATE.1.md.example (hunk 9 above)
Comment (agunde406) [reply cli/man/TEMPLATE.1.md.example]: updated

File: cli/man/TEMPLATE.1.md.example (hunk 10 above)
Comment (agunde406) [reply cli/man/TEMPLATE.1.md.example]: updated

File: cli/man/splinter-cert-generate.1.md (hunk 11 above)
Comment (agunde406) [reply cli/man/splinter-cert-generate.1.md]: updated

**Pull request merged**